### PR TITLE
[bitnami/mlflow] Fix run.initContainers render

### PR DESCRIPTION
**Pull request by fcecagno**
### Description of the change

- fix run.initContainers render
- better formatting of imagePullSecrets

### Benefits

Make it possible to use initContainers with the run component.

### Possible drawbacks

N/A

### Applicable issues

No issue created yet.

### Additional information

N/A

### Checklist

- [X] Chart version bumped in `Chart.yaml` according to [semver](http://semver.org/). This is *not necessary* when the changes only affect README.md files.
- [X] Variables are documented in the values.yaml and added to the `README.md` using [readme-generator-for-helm](https://github.com/bitnami/readme-generator-for-helm)
- [X] Title of the pull request follows this pattern [bitnami/<name_of_the_chart>] Descriptive title
- [X] All commits signed off and in agreement of [Developer Certificate of Origin (DCO)](https://github.com/bitnami/charts/blob/main/CONTRIBUTING.md#sign-your-work)

### CODE REVIEW

File: bitnami/mlflow/templates/run/dep-job.yaml [hunk 1]
```diff
@@ -100,7 +100,7 @@ spec:
         {{- end }}
         {{- end }}
         {{- if .Values.run.initContainers }}
-        {{- include "common.tplvalues.render" (dict "values" .Values.run.initContainers "context" $) | nindent 8}}
+        {{- include "common.tplvalues.render" (dict "value" .Values.run.initContainers "context" $) | nindent 8}}
```
Comment (fmulero) [inline bitnami/mlflow/templates/run/dep-job.yaml:103]: Nice catch!

File: bitnami/mlflow/templates/run/dep-job.yaml [hunk 2]
```diff
@@ -156,8 +156,9 @@ spec:
                   key: {{ include "mlflow.v0.tracking.passwordKey" . | quote }}
             {{- end }}
             {{- end }}
-            {{- if .Values.run.extraEnvVars }}
-            {{- include "common.tplvalues.render" (dict "value" .Values.run.extraEnvVars "context" $) | nindent 12 }}
+            {{- range .Values.run.extraEnvVars }}
```
Comment (fmulero) [inline bitnami/mlflow/templates/run/dep-job.yaml:159]: With this change we will loose some functionality `extraEnvVars` can be set to values like this:
```yaml
  extraEnvVars:
    - name: POSTGRESQL_CLUSTER_APP_NAME
      valueFrom:
        fieldRef:
          fieldPath: metadata.name
```
Could you share more details about the problem you are trying to solve here?

Comment (fmulero) [reply bitnami/mlflow/templates/run/dep-job.yaml:159]: Have you faced any issue setting the `extraEnvVars`?

Comment (fcecagno) [reply bitnami/mlflow/templates/run/dep-job.yaml:159]: I'll revert it, I saw it was not force-quoting the value, but I didn't face any specific issue with it.

**Comment by fcecagno** [comment]
Done @fmulero, just updated the tree, thanks for the review.

**Review by fmulero — APPROVED** [review]
Great!! Thanks @fcecagno for your contribution

**Pull request merged**